myokit/tests/test_parsing.py

```diff
@@ -1404,6 +1418,19 @@ def test_initial_values(self):
             """
         self.assertRaises(myokit.ParseError, myokit.parse, code)
 
+        # non-constant initial value
```

Comment:
(Sorry, doesn't have to be in the parser test of course, as long as its somewhere)

Reply:
an `a.b = dot(b)` would be fun too (or dot(c))

Reply:
do you expect this to parse ok? if I have:
```
[[model]]
c.p = dot(c.p)

[engine]
time = 0 bind time

[c]
dot(p) = 1
```

then `is_constant()` evaluates false for `dot(c.p)`...., what is the expected behaviour here?

Reply:
Ha!
Hadn't thought about that. I'm quite happy for it to be False because, superficially, dp/dt depends on p and t. Maybe open a ticket to investigate if that should be True in the future :D

Reply:
I _think_ I'm fine with myokit thinking dx/dt is never constant (even if the expression for it is).
It already thinks that `y/y` is an expression that depends on y, for example

Reply:
Yes. I'm sure it thinks `x = x - x` is a cyclical dependency too!

Reply:
complete now I think